polkadot/node/core/pvf/prepare-worker/src/lib.rs

```diff
@@ -168,13 +179,30 @@ pub fn worker_entrypoint(
 					Arc::clone(&condvar),
 					WaitOutcome::TimedOut,
 				)?;
+
+				let fd = stream.as_raw_fd();
+				ALLOC.start_tracking(
+					executor_params.prechecking_max_memory().map(|v| v as isize),
+					Some(Box::new(move || unsafe {
+						// Inside the failure handler, the allocator is locked and no allocations
+						// are possible
+						libc::write(
+							fd,
+							OOM_PAYLOAD as *const _ as *const libc::c_void,
+							OOM_PAYLOAD.len(),
+						);
+						libc::close(fd);
+						std::process::exit(1);
```

Comment:
It might be unlikely, but these can still technically allocate, so it's not really safe to call them from here.

(e.g. if something wrapped the `libc` functions with `LD_PRELOAD`)

The only way to do this safely is to directly call the syscalls. (Although that can only be done on Linux, so for macOS you'd have to leave it as-is and pray that it works.)